### DISCUSSION TITLE
"spin add" to add new Docker services to docker-compose yml's

### DISCUSSION THREAD

**Discussion by luchtech** [serversideup/spin#69] [Ideas]
Hi!

While sipping my coffee, I have this idea that would probably make spin more awesome. Please hear me out.

The default services when creating your first Laravel app or adding spin to your existing project are namely  **traefik**, **php**, and **mariadb**. Well, that's a very good starting point tbh. Enough for a lot of Laravel apps out there. But for long-time Laravel users, we have dabbled with more and more services that we find very useful and convenient when developing.

When I saw the contents of `docker-compose.yml` file, the urge in me to add more services was natural. So I then added **redis** for caching and queueing purposes, **meilisearch** for fuzzy search via Laravel Scout, and **minio** as S3-compatible file storage.

Don't get me wrong. I already know that there's a documentation about these services already. It's just that it would be cool if the **spin cli** has these additional commands:

```
spin add list                   # list of available services you can add to the docker-compose yml's
spin add <service>       # add a new service to the docker-compose yml's
```

I know that this is a just-good-to-have feature. But it would really improve DX since not all people are familiar with multiple docker-compose yml setup. If you like it, please share and subscribe! 😂

**Comment by harrisonratcliffe** [comment]
Yep would also like to see this so we can easily add new services.
